internal/app/server/server.go

```diff
@@ -103,6 +105,7 @@ func createServer(req *v1alpha1.ServerCompatRequest) (*http.Server, error) {
 	mux := http.NewServeMux()
 	mux.Handle(conformancev1alpha1connect.NewConformanceServiceHandler(
 		&conformanceServer{},
+		connect.WithCompression(compression.Zstd, compression.NewZstdDecompressor, compression.NewZstdCompressor),
```

Comment:
We don't have a way to specify which compression a server should support as part of `ServerCompatRequest`, but maybe we should? Otherwise, we have to specify all of them all the time, which means we can't test the scenario where a client specifies a compression algorithm the server doesn't support. 

Additionally, should we also allow for specifying a client with asymmetric compression (i.e. `WithAcceptCompression`, but not `WithSendCompression`)? Right now, it's all or nothing and the docs actually call this scenario out specifically:

> In particular, Connect encourages asymmetric compression: clients can send uncompressed requests while asking for compressed responses. ...

Reply:
> We don't have a way to specify which compression a server should support as part of `ServerCompatRequest`

This is a "supported feature". So the server should configure itself to support everything possible. And then all actual supported compression algorithms are enabled via a config file.

Reply:
> Otherwise, we have to specify all of them all the time,

I think this is fine and expected.

> which means we can't test the scenario where a client specifies a compression algorithm the server doesn't support.

This will be like having the reference client invoke an undefined entry point or sending intentionally malformed requests: only the reference client will be able to do it, and it will be done with special hand-crafted requests that don't go through the current logic. And this special request would use an intentionally non-existent compression name like "no-such-compression".

Such special requests are not yet implemented in the reference client (and special responses not yet implemented in the reference server). And I think that's totally fine for now.